src/main/java/org/craftercms/deployer/impl/processors/git/GitPullProcessor.java

```diff
@@ -120,12 +121,19 @@ protected void doPull(ProcessorExecution execution) throws DeployerException {
 
             execution.setStatusDetails(details);
         } catch (JGitInternalException e) {
-            Throwable cause = e.getCause();
-            if (cause instanceof CorruptObjectException || cause instanceof EOFException) {
+            if (isRepositoryLocked(e)) {
+                logger.warn("The local repository {} is locked, trying to unlock it", localRepoFolder);
+                try {
+                    GitUtils.unlock(localRepoFolder.getAbsolutePath());
+                    logger.info(".git/index.lock is deleted from local repository {}", localRepoFolder);
```

Comment:
Please use '{}' as a standard placeholder (I'm trying to have the whole system be consistent).